src/CoreBundle/private/core/TModelBase.class.php

```diff
@@ -194,6 +194,14 @@ public function Init()
     {
     }
 
+    /**
+     * @return bool - false if access is forbidden
+     */
+    public function checkAccessRightsOnTable(): bool
```

Comment:
I don't think the method makes sense on a module basis. For one, Permissions may or may not have anything to do with tables. But also, this is a class that is the base of frontend and backend modules - and as far as I can see, the check would not make much sense in the frontend.

For the backend we know the tables, and should look to the module definition, not the module, for checking permissions.

Reply:
(mostly this is old code refactored)

These here are the "modules" provided by TModuleLoader and they are annotated as TModelBase. And those in turn were loaded from the simple string array CmsMasterPagdef::getModuleList().

The two usecases (MTTableManager, MTTableEditor) would be TCMSModelBase but that would need to be checked by instanceof.

Reply:
Moved.

However ChameleonBackendController::checkAccess() has now a warning: Potentially polymorphic call
(wrong type of \TModuleLoader::$modules)

Reply:
Moved it back: The modules in question here are "frontend modules".

Reply:
(ie StaticViewModule or MediaManagerBackendModule)

Reply:
Moved it to its own interface: CheckTableAccessRightsInterface

However this is only an interface for two (legacy) cases: MTTableEditor, MTTableManager

For all else the access check is done via the pagedef (and only in ChameleonBackendController).